File: 0134-gas-station/0134-gas-station.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        
    int gasSum = accumulate(gas.begin(), gas.end(), 0);
    int costSum = accumulate(cost.begin(), cost.end(), 0);
    if (gasSum - costSum < 0)
      return -1;

    int result = 0;
    int sum = 0;
        
    int n = gas.size();

    for (int i = 0; i < n; ++i) {
      sum += gas[i] - cost[i];
      if (sum < 0) {
        sum = 0;
        result = i + 1;  // Start from next index.
      }
    }

    return result;
        
    }
};

File: 0134-gas-station/0134-gas-station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        
        result = 0
        difference = 0
        total = 0

        for i in range(len(gas)):
            difference += gas[i] - cost[i]
            total += gas[i] - cost[i]
            if total < 0:
                total = 0
                result = i + 1  # Start from next i

        return -1 if difference < 0 else result